File: hw/villagomez/u2/hw18ZooJS/Zoo.js
//const { Animal, Lion, Monkey, Shark, Cage, Food } = require('./ec.edu.espe.zoo.model'); 
const Lion = require('./ec.edu.espe.zoo.model/Lion');
const Monkey = require('./ec.edu.espe.zoo.model/Monkey');
const Shark = require('./ec.edu.espe.zoo.model/Shark');
const Condor = require('./ec.edu.espe.zoo.model/Condor');
const Eagle = require('./ec.edu.espe.zoo.model/Eagle');
const Cage = require('./ec.edu.espe.zoo.model/Cage');
const Food = require('./ec.edu.espe.zoo.model/Food');

const animals = [];

console.log("Domenica's Zoo");

let cageFelines = new Cage(1, "FelinesCage");
let cageMonkeys = new Cage(2, "monkeysCage");
let cageBirds = new Cage(3, "BirdsCage");
let pool = new Cage(4, "Pool for Shark");

let food = new Food(1, "Fish");

const lion = new Lion(true, 1, "Lion Patricio", new Date(2024, 10, 10), 'f', 0, cageFelines);
lion.register(cageFelines);
animals.push(lion);
lion.feed(2);
lion.feed(3, food);
console.log("Lion --> ", lion);

let animal;

animal = new Monkey(2, "Furry", new Date(2024, 5, 3), 'f', 0, cageMonkeys);
animal.feed(5);
animals.push(animal);
console.log("Animal (monkey) --> ", animal);

animal = new Lion(false, 2, "Lion Percy", new Date(2024, 5, 3), 'f', 0, cageFelines);
animal.feed(4);
animals.push(animal);
console.log("Animal (lion) --> ", animal);

const shark = new Shark(1, "Dad Shark", new Date(2024, 6, 7), 'f', 0, pool);
animal.feed(6);
animals.push(shark);
console.log("Animal (shark) --> ", shark);

console.log("Animals in Domenica's Zoo --> \n", animals);

for (let animal of animals) {
    console.log(animal);
}

File: ws/gualotuna/u2/WS24Zoo/Zoo/src/main/java/ec/edu/espe/zoo/view/Zoo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.zoo.view;

import ec.edu.espe.zoo.model.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Date;


/**
 *
 * @author Brayan Gualotuña, Dev Dynasty, DCCO-ESPE
 */
public class Zoo {
    public static void main(String[] args) {
        System.out.println("Brayan's Zoo");
        Cage cageFelines = new Cage(1,"felineCage");
        Cage cageMonkeys = new Cage(2, "monkeysCage");
        Cage cageBirds = new Cage(3, "BirdsCage");
        Cage pool = new Cage(4,"pool of Sharks");
        Food food = new Food(1, "fish");
        ArrayList<Animal> animals = new ArrayList<>();
        
        
        Lion lion = new Lion(true, 1, "Lion Juancho", new Date(2000, 2, 3), 'm', cageFelines, 0, 0);
        lion.register();
        lion.feed(2);
        lion.feed(3, food);
        System.out.println("lion --->" + lion);
       
        
        Animal animal;
        //animal = new Animal();
        
        animal = new Monkey(2, "bear", new Date(2000, 2, 3), 'm', cageMonkeys,0 , 0);
        animal.feed(5);
        animal.register();
        
        
        
        animal.feed(4);
        
        Shark shark = new Shark(4, "baby shark", new Date(2000, 2, 3), 'm', pool, 0, 0);
        shark.feed(1);
        //animals.add(shark);
        System.out.println("Shark ---> " + shark);
        
        System.out.println("animals in Brayan's Zoo --->" + animals);
        
        for (Animal ani : animals) {
            System.out.println(ani);
        }
        
    }
}
